jupyter_ascending/handlers/server_extension.py

```diff
@@ -0,0 +1,131 @@
+from pathlib import Path
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+from jsonrpcclient import request
+from jsonrpcserver import dispatch
+from jsonrpcserver import method
+from notebook.base.handlers import IPythonHandler  # type: ignore
+from notebook.utils import url_path_join  # type: ignore
+
+from jupyter_ascending._environment import SYNC_EXTENSION
+from jupyter_ascending.errors import UnableToFindNotebookException
+from jupyter_ascending.functional import get_matching_tail_tokens
+from jupyter_ascending.logger import J_LOGGER
+
+_REGISTERED_SERVERS: Dict[str, int] = {}
+
+
+def _clear_registered_servers():
+    global _REGISTERED_SERVERS
+
+    _REGISTERED_SERVERS = {}
+
+
+class JupyterAscendingHandler(IPythonHandler):
+    # def get(self):
+    #     print("got get request!")
+    #     self.finish('Hello, world!')
+
+    def post(self) -> None:
+        """We receive commands as HTTP POST requests.
+
+        TODO: might want to convert this to async in the future
+        (tornado and jsonrpcserver support asyncio)"""
```

Comment:
Can remove this TODO until someone needs it probably, unless there is a reason to do it now?

Reply:
Hmm, I left that there because while it seems to perform fine now, if we have performance issues in the future this is something someone might wanna look into - and they might not know these libraries support asyncio easily without a note like this.